xcbgen-rs/src/defs.rs

```diff
@@ -772,13 +1221,17 @@ impl VariableSize {
                 incr: self.incr,
             }
         } else {
+            // FIXME: Why is it correct to just ignore `incr` in this case?
```

Comment:
The equation here would be:

`(base + incr * n) * m = base * m + incr * n * m = gcd(base, incr) * l`

Reply:
What do you think of https://github.com/psychon/x11rb/commit/b5540185fcfd46c330644f9b7d7fa78c05d1c776 ?

Reply:
The last step is not really an equality, but rather something like "all values of the left side are always a multiple of the right side", but whatever.

> What do you think of b554018 ?

Uhm, so `1 << (a | b).trailing_zeros() = gcd(a,b)` if `b` is a power of two... okay.... uhm....

We have `b=2^c` for some `c` and `a=2^d*o` for some `c` and odd `o`. The GCD is now `2^min(c,d)`. `b` has `c` trailing zeros and `a` has `d` trailing zeros, so `a|b` has `min(c,d)` trailing zeros. After some thinking I get this.

Reply:
All this alignment thing is a bit tricky...